git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,503 @@
+</div>
+
+Now that we have found the identifier of the commit we want to examine, we can use it with `git show`. Don't worry, you won't need to type all 40 digits, the first six will suffice.
+
+```
+$ git show 1e587d
+commit 1e587d25f619aa0aa10fce19b44e4e71503fa41e
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Wed Apr 6 13:13:21 2022 -0400
+
+    Add concerns about effects of Mars' moons on Wolfman
+
+diff --git a/mars.txt b/mars.txt
+index df0654a..315bf3a 100644
+--- a/mars.txt
++++ b/mars.txt
+@@ -1 +1,2 @@
+ Cold and dry, but everything is my favorite color
++The two moons may be a problem for Wolfman
+```
+
+Unlike `HEAD`, the commit number doesn't change or move as you update your repository.
+
+![On the left is the stack of 3 boxes, each labeled with its six digit commit number. An arrow points from this stack to the stack on the right where the three white boxes have a red box on top. The red box also has a commit number.  The commit numbers on the white boxes have not changed from the picture on the left.](media/Commit_number.svg)
+
+### Quiz: finding a commit
+
+The output from `git log -n 2` is:
+
+```
+$ git log -n 2
+commit 024af1d5d311c5d1a737cca3fb13d1b62c2a29ed (HEAD -> main, origin/main)
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Wed Apr 6 13:17:26 2022 -0400
+
+    Ignore data files and the results folder
+
+commit 656471fa1f38c8b0b9a6a2fafdf3ba68610e61fa
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Wed Apr 6 13:15:24 2022 -0400
+
+    Start files on Jupiter and Mercury
+```
+
+Which of the following commands would show you the **most recent** commit you made? Select all correct answers.
+
+- [[X]] `git show HEAD`
+- [[ ]] `git show HEAD~1`
+- [[X]] `git show 024af1`
+- [[ ]] `git show 2a29ed`
+***
+<div class ="answer">
+As long as you didn't do anything fancy to move `HEAD`, the **most recent** commit is the current `HEAD`. Its commit number is `584977`. Since both refer to the same commit, you can use either.
+
+The command `git show HEAD~1` will show you one checkpoint earlier in your work, while `git show 081cd9` will give you an error since `081cd9` it is not the **first** six digits of a known commit number.
+</div>
+***
+
+<div class = "care">
+The error you get from entering a non-existent commit number starts with the word "fatal."  The word may be scary, but, in this case, all it is saying is that it could not execute the command. You did not harm anything!
+</div>
+
+## Comparing files to prior commits
+
+In order to have some more commits to compare, let's add a few more lines to Dracula's notes about Mars.
+
+First add a line to `mars.txt` so that it reads:
+
+```
+Cold and dry, but everything is my favorite color
+The two moons may be a problem for Wolfman
+But the Mummy will appreciate the lack of humidity
+```
+Let's add and commit this change:
+
+```
+$ git add mars.txt
+$ git commit -m "Discuss pros of Martian climate for Mummy"
+```
```

Comment:
Is there a way in Liascript to have a text box that includes a prompt on each line, so you can make the whole box copy/paste?  Copying one line at a time to avoid getting that $ is annoying, at least to me.  Or maybe omit the $?  I'm not sure if there's a style guide we can crib from as far as other education is concerned... how do other educators handle this? hmmmmmm

Reply:
My understanding (not sure from where) is that this is mainly to distinguish input lines from output lines in when they are in the same code block. But it is completely possible that it was standard in the academic CS circles I have been in specifically to _discourage_ copying and pasting, when we may well want to encourage copy/pasting. 

Things to think about: If we remove the `$`s from code blocks that don't have any output, do we still keep them in code blocks with both input and output? Do we want to do some fancy `css` stuff to make input vs. output of code blocks clearer?